index.jsx

```diff
@@ -1,15 +1,27 @@
 import React, {PropTypes} from 'react';
 
-function radio(name, selectedValue, onChange) {
+function radio(kwargs) {
+
   return React.createClass({
     render: function() {
+      const name = kwargs.name;
+
+      // We only want to assign these attribute to the <input />
+      // elements if they were specified explicitly.
+      const optional = {};
+      if(kwargs.hasOwnProperty("selectedValue")) {
```

Comment:
> "No. React should treat this as an Uncontrolled Field when selectedValue={undefined}"

That is what I think. Undefined is undefined, it does not exist. I can be wrong and React itself does check hasOwnProperty and not `undefined`. If you keen, would be nice to double-check React's source.